### PR TITLE
offset-distance parse and merge with interpolation tests 

### PR DESCRIPTION
**Pull request by courtneycb**
Has magic numbers, not sure how to get around using them.

### CODE REVIEW

File: src/offsetDistance.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+       initial value is 0 so will return 0 on undefined.
+       Will return a number when units is px and a number array if units is %.
+    */
+    if (input === undefined) {
+      return 0;
+    }
+
+    var distance = 0;
+
+    if (input.charAt(input.length - 1) === '%') {
```
Comment (ewilligers) [inline src/offsetDistance.js]: input.endsWith('%')

File: src/offsetDistance.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      distance = input.substring(0, input.length - 1);
+      if (!(isNumeric(distance))) {
+        // distance must be a number
+        return 0;
+      }
+      return [Number(distance)];
+    }
+
+    // see if unit is px
+    var unit = input.substring(input.length - 2, input.length);
```
Comment (ewilligers) [inline src/offsetDistance.js]: if (!input.endsWith('px'))

This way, we avoid index out of bounds issues if input has length less than 2.

File: src/independentTransforms.js [hunk 3]
```diff
@@ -15,14 +15,18 @@
...
+        merge: internalScope.offsetDistanceMerge
       }
     },
     applyHook: {
       callback: function (values) {
         var transformString = internalScope.toTransform(values);
         return {transform: transformString + ' ' + values.transform};
       },
-      watchedProperties: ['scale', 'rotate', 'translate', 'transform']
+      watchedProperties: ['scale', 'rotate', 'translate', 'transform', 'offset-distance']
```
Comment (alancutter) [inline src/independentTransforms.js]: No need to add to the watched properties if it's not being used yet.

File: src/offsetDistance.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    if (unit !== 'px' || !(isNumeric(distance))) {
+      // unit must be one of px or % and distance must be a number
+      return 0;
+    }
+    return Number(distance);
+  }
+
+  function offsetDistanceMerge (start, end) {
+    function serialize (input) {
+      if (typeof input === 'number') {
```
Comment (alancutter) [inline src/offsetDistance.js]: This is a really awkward way of representing the unit type. Use something like {value: 20, unit: 'px'} instead.

File: src/offsetDistance.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        return input + 'px';
+      }
+      return input[0] + '%';
+    }
+
+    if (typeof start !== typeof end) {
+      return {
```
Comment (ewilligers) [inline src/offsetDistance.js]: return flip(serialize(start), serialize(end));

where flip is a new helper.

File: src/fileOrder.txt [hunk 6]
```diff
@@ -1,5 +1,6 @@
 src/rotate.js
```
Comment (ewilligers) [inline src/fileOrder.txt]: We could add a helpers.js at the start

which would contain isNumeric and flip.

**Pull request merged**